kroxylicious-api/src/main/java/io/kroxylicious/proxy/ApiMaturity.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotations to specify the maturity of APIs.
+ */
+public class ApiMaturity {
+
+    /**
+     * No guarantee is provided as to reliability or stability across any level of release granularity.
+     *
+     * If an element is not annotated with an ApiMaturity annotation then the maturity is implicitly the maturity of
+     * the enclosing declaration.
+     */
+    @Target({ ElementType.TYPE, ElementType.CONSTRUCTOR, ElementType.METHOD, ElementType.ANNOTATION_TYPE, ElementType.PACKAGE })
+    @Retention(RetentionPolicy.SOURCE)
+    @Documented
+    public @interface Unstable {
+    }
+
+    /**
+     * Compatibility may be broken at minor release (i.e. m.x).
+     *
+     * If an element is not annotated with an ApiMaturity annotation then the maturity is implicitly the maturity of
+     * the enclosing declaration.
+     */
+    @Target({ ElementType.TYPE, ElementType.CONSTRUCTOR, ElementType.METHOD, ElementType.ANNOTATION_TYPE, ElementType.PACKAGE })
+    @Retention(RetentionPolicy.SOURCE)
+    @Documented
+    public @interface Evolving {
+    }
+
+    /**
+     * Compatibility is maintained in major, minor and patch releases with one exception: compatibility may be broken
+     * in a major release (i.e. 0.m) for APIs that have been deprecated for at least one major/minor release cycle.
```

Comment:
I'm not quite sure how to read `at least one major/minor release cycle.` 

Does it mean that a the old method gets deprecated in 0.4 are removed 0.5? Or 0.6 (i.e. they survive for one full minor cycle)? Or do you mean deprecated in 0.4 and can only be removed in 1.0 ? 

I think an examples table would be in order here.

Reply:
This is my reading.

## release timeline 1:
1. 0.4 deprecates methodA
2. 0.5 release made
3. 1.0 release made - removes methodA

## release timeline 2
1. 0.4 deprecates methodA
2. 1.0 release made - cannot remove methodA
3. 1.1 release made - still cannot remove methodA
4. 2.0 release made - methodA removed